hydroflow/tests/surface_lattice_join.rs

```diff
@@ -0,0 +1,130 @@
+use hydroflow::util::collect_ready;
+use hydroflow_macro::hydroflow_syntax;
+use multiplatform_test::multiplatform_test;
+
+#[multiplatform_test]
+pub fn test_lattice_join_reducing_behavior() {
+    // lattice_fold has the following particular initialization behavior. It uses LatticeFrom to convert the first element into another lattice type to create the accumulator and then it folds the remaining elements into that accumulator.
+    // This initialization style supports things like SetUnionSingletonSet -> SetUnionHashSet. It also helps with things like Min, where there is not obvious default value so you want reduce-like behavior.
+    let mut df = hydroflow_syntax! {
+        use lattices::Min;
+        use lattices::Max;
+
+        source_iter([(7, Min::new(5)), (7, Min::new(6))]) -> [0]my_join;
+        source_iter([(7, Max::new(5)), (7, Max::new(6))]) -> [1]my_join;
+
+        my_join = lattice_join::<Min<usize>, Max<usize>>()
+            -> assert_eq([(7, (Min::new(5), Max::new(6)))]); // If Min::default is used, then this would likely produce (7, (0, 6)), which is incorrect.
```

Comment:
https://github.com/hydro-project/hydroflow/issues/863

Reply:
I'll remove the comment but keep the test.